src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/AccessEgressMapper.java

```diff
@@ -23,7 +22,9 @@ public AccessEgressMapper(StopIndexForRaptor stopIndex) {
   public AccessEgress mapNearbyStop(NearbyStop nearbyStop, boolean isEgress) {
     if (!(nearbyStop.stop instanceof Stop)) { return null; }
     return new AccessEgress(
-        stopIndex.indexByStop.get(nearbyStop.stop),
+        // TODO: This cast is potentially causing problems look into if it is possible
```

Comment:
We have an `instanceof` call above, so this is safe. Also, the `stopsByIndex` uses the wider type after #3757, so it might be possible to clean this up.